Sources/Regex/Parse/CharacterPropertyClassification.swift

```
@@ -0,0 +1,392 @@
+    }
+  }
+
+  static func classifyCharacterPropertyValueOnly(
+    _ value: String
+  ) throws -> Atom.CharacterProperty.Kind {
+    guard !value.isEmpty else { throw ParseError.emptyProperty }
+
+    // Some special cases defined by UTS#18 (and Oniguruma for 'ANY' and
+    // 'Assigned').
+    switch normalizeForCharacterPropertyMatching(value) {
+    case "any": return .any
+    case "assigned": return .assigned
+    case "ascii": return .ascii
+    default: break
+    }
+
+    // The following properties we can infer keys/values for.
+    if let prop = classifyBoolProperty(value) {
+      return .binary(prop, value: true)
+    }
+    if let cat = classifyGeneralCategory(value) {
+      return .gc(cat)
+    }
+    if let script = classifyScriptProperty(value) {
+      return .script(script)
+    }
+
+    // Some additional special cases we recognise.
+    // TODO: Normalize these?
+    if let oniguruma = OnigurumaSpecialProperty(rawValue: value) {
+      return .onigurumaSpecial(oniguruma)
+    }
+    typealias PCRESpecial = Atom.CharacterProperty.Kind.PCRESpecialCategory
+    if let pcreSpecial = PCRESpecial(rawValue: value) {
+      return .pcreSpecial(pcreSpecial)
+    }
+
+    // Otherwise we don't know what this is.
+    return .other(key: nil, value: value)
+  }
+
+  static func classifyCharacterProperty(
+    key: String, value: String
+  ) throws -> Atom.CharacterProperty.Kind {
+    guard !key.isEmpty && !value.isEmpty else { throw ParseError.emptyProperty }
+
+    if let prop = classifyBoolProperty(key) {
+      let isTrue = try classifyCharacterPropertyBoolValue(value)
+      return .binary(prop, value: isTrue)
+    }
+
+    // This uses the aliases defined in
+    // https://www.unicode.org/Public/UCD/latest/ucd/PropertyAliases.txt.
+    switch normalizeForCharacterPropertyMatching(key) {
+    case "script", "sc":
+      guard let script = classifyScriptProperty(value) else {
+        throw ParseError.invalidScriptKind(value)
+      }
```

@milseman Any thoughts on whether we should throw an error here, or just fallback to parsing as an `.other` in case new scripts (and maybe general categories?) get added in the future?

---

Run time can have a different version of the spec which could be newer. I don't know how we'll handle that kind of availability... We may need to communicate back to the parser and our enum/struct or whatever has availability on it.

---

Okay yeah, I've updated to parse unknown cases as `other` for now